src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexWriter.cs

```diff
@@ -109,7 +132,7 @@ private int CurPos()
         /// </summary>
         private void PatchJump(int offset, int jumpDest)
         {
-            _emitted[offset + 1] = jumpDest;
+            Emitted[offset + 1] = jumpDest;
```

Comment:
can`t builder just be passed as ref here?

Reply:
I tried that approach which resulted in all methods requiring 2-3 more parameters. Passing them around degrades the codes readability a lot. The initial design of this class was to have a static entry point `RegexWriter.Write` which then creates a private instance of the class just to make it easier to share fields. I'm not huge fan of that either as it involves allocation of an unnecessary object on the heap (it was a class before).

Reply:
Passing the ValueBuilders as ref degrades performance substantially because of the passing of refs (just tried that, costs doubled).